adapters/audienceNetwork/facebook.go

```diff
@@ -37,267 +41,416 @@ func (a *FacebookAdapter) Name() string {
 	return "audienceNetwork"
 }
 
-// Facebook likes to parallelize to minimize latency
-func (a *FacebookAdapter) SplitAdUnits() bool {
-	return true
-}
-
 func (a *FacebookAdapter) SkipNoCookies() bool {
 	return false
 }
 
-type facebookParams struct {
-	PlacementId string `json:"placementId"`
+type facebookReqExt struct {
+	PlatformId string `json:"platformid"`
+	AuthId     string `json:"authentication_id"`
 }
 
-func coinFlip() bool {
-	return rand.Intn(2) != 0
-}
+func (this *FacebookAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "no impressions provided",
+		}}
+	}
 
-func (a *FacebookAdapter) callOne(ctx context.Context, reqJSON bytes.Buffer) (result adapters.CallOneResult, err error) {
-	url := a.URI
-	if coinFlip() {
-		//50% of traffic to non-secure endpoint
-		url = a.nonSecureUri
+	if request.User == nil || request.User.BuyerUID == "" {
+		return nil, []error{&errortypes.BadInput{
+			Message: "Missing bidder token in 'user.buyeruid'",
+		}}
 	}
-	httpReq, _ := http.NewRequest("POST", url, &reqJSON)
-	httpReq.Header.Add("Content-Type", "application/json")
-	httpReq.Header.Add("Accept", "application/json")
-
-	anResp, e := ctxhttp.Do(ctx, a.http.Client, httpReq)
-	if e != nil {
-		err = e
-		return
+
+	reqs, errs := this.buildRequests(request)
+	return reqs, errs
```

Comment:
You can return the result of the `buildRequests` call directly. No need to assign local variables.

Reply:
done